lib/ansible/module_utils/network/common/netconf.py

```diff
@@ -67,25 +67,28 @@ def __rpc__(self, name, *args, **kwargs):
 
     def parse_rpc_error(self, rpc_error):
         if self.check_rc:
-            error_root = fromstring(rpc_error)
-            root = Element('root')
-            root.append(error_root)
-
-            error_list = root.findall('.//nc:rpc-error', NS_MAP)
-            if not error_list:
-                raise ConnectionError(to_text(rpc_error, errors='surrogate_then_replace'))
-
-            warnings = []
-            for error in error_list:
-                try:
-                    message = error.find('./nc:error-message', NS_MAP).text
-                except Exception:
-                    message = error.find('./nc:error-info', NS_MAP).text
-
-                severity = error.find('./nc:error-severity', NS_MAP).text
-
-                if severity == 'warning' and self.ignore_warning:
-                    warnings.append(message)
-                else:
+            try:
+                error_root = fromstring(rpc_error)
+                root = Element('root')
+                root.append(error_root)
+
+                error_list = root.findall('.//nc:rpc-error', NS_MAP)
+                if not error_list:
                     raise ConnectionError(to_text(rpc_error, errors='surrogate_then_replace'))
-            return warnings
+
+                warnings = []
+                for error in error_list:
+                    try:
+                        message = error.find('./nc:error-message', NS_MAP).text
+                    except Exception:
+                        message = error.find('./nc:error-info', NS_MAP).text
+
+                    severity = error.find('./nc:error-severity', NS_MAP).text
+
+                    if severity == 'warning' and self.ignore_warning:
+                        warnings.append(message)
+                    else:
+                        raise ConnectionError(to_text(rpc_error, errors='surrogate_then_replace'))
+                return warnings
+            except XMLSyntaxError:
+                raise ConnectionError(rpc_error)
```

Comment:
Guess as `xml.tree` doesn't have `XMLSyntaxError `, a catch all `except:` would be good here?

Reply:
So we're importing three different exceptions to that name, depending on context, but I think we should have everything covered now